### PR TITLE
remove image podman no prune

### PR DESCRIPTION
**Pull request by karthikelango137**
Docker supports --no-prune for rmi which does not remove dangling parents of the specified image. Podman
should support the same for compatibility.

Task from #14917 

Signed-off-by: Karthik Elango <kelango@redhat.com>

#### Does this PR introduce a user-facing change?

```release-note
Added --no-prune flag for podman rmi
```

### CODE REVIEW

File: go.mod
Comment (vrothberg) [inline go.mod:78]: Just to be extra sure.  Before merging, we need a PR in c/common and then vendor from there.

**Comment by karthikelango137** [comment]
@containers/podman-maintainers PTAL

**Review by ashley-cui — COMMENTED** [review]
LGTM, but looks like you need a rebase or something, as you have a merge conflict.

File: go.sum
Comment (cdoern) [inline go.sum:398]: leftover rebase weirdness

File: vendor/modules.txt
Comment (cdoern) [inline vendor/modules.txt:117]: this looks like a downgrade

**Review by cdoern — COMMENTED** [review]
some vendor/rebase leftover stuff, otherwise LGTM

**Review by vrothberg — COMMENTED** [review]
LGTM
@containers/podman-maintainers PTAL

Nice work, @karthikelango137 !

**Review by ashley-cui — COMMENTED** [review]
/lgtm

Great job @karthikelango137 ! :)

**Review by Luap99 — APPROVED** [review]
LGTM

**Pull request merged**
